### PR TITLE
Fix "xstructure method not found" error in certain 404 conditions

### PR DESCRIPTION
**Pull request by linc**
* Add missing views for 3 actions (remove picture, delete category photo, delete activity).
* Remove janky Home controller faking & FileNotFound view calling.
* Fix bad redirect checking on `ActivityController->Delete()`.

See #3179

### CODE REVIEW

File: applications/dashboard/controllers/class.activitycontroller.php [hunk 1]
```diff
@@ -237,13 +237,13 @@ public function delete($ActivityID = '', $TransientKey = '') {
...
+        $target = Gdn::request()->get('Target');
+        if ($this->_DeliveryType === DELIVERY_TYPE_ALL && $target) {
+            redirect($target);
         }
 
-        // Still here? Getting a 404.
-        $this->ControllerName = 'Home';
-        $this->View = 'FileNotFound';
+        // Still here? Getting a basic view with redirect.
+        $this->RedirectUrl = url('activity');
```
Comment (beckyvb) [inline applications/dashboard/controllers/class.activitycontroller.php]: This redirect causes the page to reload after deleting an activity item. Is it necessary?

File: applications/vanilla/controllers/class.settingscontroller.php [hunk 2]
```diff
@@ -466,13 +466,16 @@ public function deleteCategoryPhoto($CategoryID = false, $TransientKey = '') {
...
             redirect($RedirectUrl);
         } else {
-            $this->ControllerName = 'Home';
-            $this->View = 'FileNotFound';
             $this->RedirectUrl = url($RedirectUrl);
             $this->render();
         }
     }
 
+    /**
```
Comment (nervoustwit) [inline applications/vanilla/controllers/class.settingscontroller.php]: WWLS, What would Lincoln say? This doc block could use a description.

File: applications/vanilla/controllers/class.settingscontroller.php [hunk 3]
```diff
@@ -466,13 +466,16 @@ public function deleteCategoryPhoto($CategoryID = false, $TransientKey = '') {
...
+     *
+     *
+     * @param $Category
+     */
```
Comment (nervoustwit) [inline applications/vanilla/controllers/class.settingscontroller.php]: WWLS, What would Lincoln say? This doc block could use a description.

Comment (linc) [inline applications/vanilla/controllers/class.settingscontroller.php]: I didn't even add that method! Low blow. Blank block > no block.

File: applications/dashboard/controllers/class.activitycontroller.php (hunk 1 above)
Comment (linc) [inline applications/dashboard/controllers/class.activitycontroller.php]: No; the concern was that if, thru a failure of Javascript, you manage to actually land on the delete activity page, you'll then be stuck there. Obviously that doesn't justify a crappy default experience.

Comment (linc) [inline applications/dashboard/controllers/class.activitycontroller.php]: Another check for DELIVERY_TYPE_ALL here will fix, I think.

File: applications/vanilla/controllers/class.settingscontroller.php (hunk 3 above)
Comment (nervoustwit) [inline applications/vanilla/controllers/class.settingscontroller.php]: Touché.

**Comment by linc** [comment]
@beckyvb look better?

**Comment by msayle** [comment]
I completed this fix and it appears now I can add comments on my profile but there is nothing allowing me to delete them. 

No button, no gear icon, nothing but the 'comment' button.

**Pull request merged**